pkg/cmd/server/api/secretstring_test.go

```diff
@@ -0,0 +1,177 @@
+package api
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+var (
+	// encrypted data generated by `oadm ca encrypt`
+	// if the output of `oadm ca encrypt` changes, add additional tests,
+	// but do not modify this, to ensure we continue to decrypt these values
+	encryptedData = []byte(`-----BEGIN ENCRYPTED STRING-----
+Proc-Type: 4,ENCRYPTED
+DEK-Info: AES-256-CBC,8db8d0db237a56c8ea8d3c5c21acc5b6
+
+9JUj8ezx/3MDgiiWbnx9QA==
+-----END ENCRYPTED STRING-----`)
+
+	// encrypting key generated by `oadm ca encrypt`
+	// if the output of `oadm ca encrypt` changes, add additional tests,
+	// but do not modify this, to ensure we continue to decrypt these values
+	encryptingKey = []byte(`-----BEGIN ENCRYPTING KEY-----
+f3zQfReuhwI1BvBNglhZdjgSocKKqABwyGafHJcdORw=
+-----END ENCRYPTING KEY-----`)
+)
+
+func TestExternalizableString(t *testing.T) {
+	os.Setenv("present_env", "envvalue")
+	os.Setenv("encrypted_env", string(encryptedData))
+
+	emptyFile, err := ioutil.TempFile("", "empty_file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(emptyFile.Name()) // clean up
+
+	fooFile, err := ioutil.TempFile("", "foo_file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(emptyFile.Name()) // clean up
+	if err := ioutil.WriteFile(fooFile.Name(), []byte(`filevalue`), os.FileMode(0755)); err != nil {
+		t.Fatal(err)
+	}
+
+	encryptedFile, err := ioutil.TempFile("", "encrypted_file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(encryptedFile.Name()) // clean up
+	if err := ioutil.WriteFile(encryptedFile.Name(), encryptedData, os.FileMode(0755)); err != nil {
+		t.Fatal(err)
+	}
+
+	validKeyFile, err := ioutil.TempFile("", "valid_key_file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(validKeyFile.Name()) // clean up
+	if err := ioutil.WriteFile(validKeyFile.Name(), encryptingKey, os.FileMode(0755)); err != nil {
+		t.Fatal(err)
+	}
+
+	invalidKeyFile, err := ioutil.TempFile("", "invalid_key_file")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(invalidKeyFile.Name()) // clean up
+	if err := ioutil.WriteFile(invalidKeyFile.Name(), []byte(`invalid key value`), os.FileMode(0600)); err != nil {
+		t.Fatal(err)
+	}
+
+	testcases := map[string]struct {
+		ExternalizableString ExternalizableString
+		ExpectedValue        string
+		ExpectedError        bool
```

Comment:
I've found `ExpectedError string`  where the string is a substring of the expected error message keeps us from passing the unit test while failing the actual scenario.